analyses/focal-cn-file-preparation/05-define-most-focal-cn-units.Rmd

```diff
@@ -0,0 +1,213 @@
+
+```{r}
+results_dir <- "results"
+
+# Define a logical object for running in CI
+running_in_ci <- params$is_ci
+```
+
+### Read in files
+
+```{r}
+# Read in the file with consensus CN status data and the UCSC cytoband data --
+# generated in `03-add-cytoband-status-consensus.Rmd`
+consensus_seg_cytoband_status_df <-
+  read_tsv(file.path("results", "consensus_seg_with_ucsc_cytoband_status.tsv.gz"))
+
+# Read in the annotated CN file (without the UCSC data)
+consensus_seg_autosomes_df <-
+  read_tsv(file.path(results_dir, "consensus_seg_annotated_cn_autosomes.tsv.gz"))
+```
+
+## Define most focal units
+
+### First, we want find entire chromosome arm losses/gains.
+
+```{r}
+consensus_seg_cytoband_status_df <-
+  consensus_seg_cytoband_status_df %>%
+  select(chromosome_arm,
+         dominant_status,
+         cytoband,
+         Kids_First_Biospecimen_ID)
+
+arm_status_count <- consensus_seg_cytoband_status_df %>%
+  # Count the amount og times each arm has a specific `dominant_status` value
+  count(chromosome_arm, dominant_status) %>%
+  # Spread the data -- each row represents a unique chromosome arm
+  spread(dominant_status, n) %>%
+  select(-c("uncallable", "unstable")) %>%
+  replace_na(list(
+    gain = 0,
+    loss = 0,
+    neutral = 0
+  )) %>%
+  group_by(chromosome_arm) %>%
+  # The logic here is to define the arm status based on the majority of calls
+  # in the chromosome arm -- if the number of calls for a specific status is
+  # responsible for more than 50% of the total calls in that arm, then it is
+  # used to define the arm status
+  mutate(
+    arm_dominant = case_when(
+      gain > (0.49 * (gain + loss + neutral)) ~ "arm_gain",
+      loss > (0.49 * (gain + loss + neutral)) ~ "arm_loss",
+      neutral > (0.49 * (gain + loss + neutral)) ~ "arm_neutral",
+      TRUE ~ "uncallable"
+    )
+  ) %>%
+  select(c("chromosome_arm", "arm_dominant"))
+
+# Display table
+arm_status_count
+```
+
+### Next we want to use the annotated CN data to define the cytoband gains/losses 
+
+We want to include cytoband and gene level calls for chromosome arms that have not been defined as a gain or loss.
+
+```{r}
+# These are the chromosome arms that have not been defined as gain or loss --
+# we want to define their cytoband/gene level status
+neutral_arm_status_df <- arm_status_count %>%
+  filter(arm_dominant == "arm_neutral")
+
+# Filter the annotated CN data to include only these chromosome arms
+consensus_seg_autosomes_df_develop <- consensus_seg_autosomes_df %>%
+  select(biospecimen_id, gene_symbol, cytoband, status) %>%
+  mutate(chromosome_arm = gsub("(p|q).*", "\\1", cytoband)) %>%
+  filter(chromosome_arm %in% neutral_arm_status_df$chromosome_arm)
+
+# Now count the cytoband level calls (for each status call) and define
+# the cytoband as that status if more than 50% of the total counts are
+# for that particular status
+cytoband_status_count <- consensus_seg_autosomes_df_develop %>%
+  count(cytoband, status) %>%
```

Comment:
Looks like this section is the same as lines 67 - 89,  may want to make it a function? But we should hammer down the logic for majority calls.